rfcs/59-from_reflect_ergonomics.md

```diff
@@ -0,0 +1,284 @@
+# Feature Name: `from_reflect_ergonomics`
+
+## Summary
+
+Derive `FromReflect` by default when deriving `Reflect`, add `ReflectFromReflect` type data, and make `ReflectDeserializer` easier to use.
+
+## Terminology
+
+#### *Dynamic Types*
+
+To make this document easier to read, the term "**Dynamic**" (with this casing) will be used to refer to any of the `Dynamic***` structs. This includes `DynamicStruct`, `DynamicTupleStruct`, `DynamicList`, etc.
+
+Other forms might include "Dynamics," "Dynamic structs," or "Dynamic types" depending on context.
+
+In example code, variables containing these types will be denoted as `dyn_X`, where `X` corresponds to their data structure. For example, `dyn_tuple_struct` would represent a `DynamicTupleStruct`.
+
+#### *Real Types*
+
+The term "**real type**" will be used to distinguish Dynamics and other concrete types. We can define it as any concrete type known at compile time that is *not* one of the Dynamic structs. 
+
+In example code, variables containing real types will be denoted as `real_Y`, where `Y` corresponds to their data structure. For example, `real_list` would represent a list-like type, such as `Vec<T>`.
+
+## Background
+
+### What is `FromReflect`?
```

Comment:
I think we might want to start even earlier, with "what is `Reflect`". Remember that the audience is the average Bevy developer, who hasn't touched reflection at all.

If we had good introductory module docs (or those on `Reflect`), I'd link those, but we don't.